my-app/src/components/login/LogInComponent.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
+import { Redirect } from "react-router-dom";
+import { login, isAuthorized } from "../../services/auth.service";
+import constantsService from "../../constants/constants.service";
+
+export default class LogInComponent extends Component {
+  state = {
+    isAuthorized: isAuthorized(), // on init
+    error: null,
+    isLoaded: false,
+    username: "",
+    password: "",
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    login({
+      username: this.state.username,
+      password: this.state.password,
+    })
+      .then((res) => {
+        this.setState({ isAuthorized: isAuthorized() });
+      })
+      .catch((error) => {
+        alert(error.message);
+      });
+  };
+
+  onChangeInput = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    this.setState({ [name]: value });
+  };
+
+  render() {
+    const { isAuthorized } = this.state;
+    // = const isAuthorized = this.state.isAuthorized;
+    return isAuthorized ? (
```

Comment:
Why do you still use isAuthorised here, if you have protected routes

Reply:
yes, yes, I'll try to figure it out) Thanks for pointing it out